File: cpp/difference-of-squares/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace squares
{
	const int square_of_sums(const int& n)
	{
		int s = 0;
		for (int i = 1; i <= n; i++)
		{
			s += i;
		}
		return s * s;
	}

	const int sum_of_squares(const int& n)
	{
		int s = 0;
		for (int i = 1; i <= n; i++)
		{
			s += i * i;
		}
		return s;
	}

	const int difference(const int& n)
	{
		return square_of_sums(n) - sum_of_squares(n);
	}
}

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSums(n int) int {
	s := 0
	for i := 1; i <= n; i++ {
		s += i
	}
	return s * s
}

func SumOfSquares(n int) int {
	r := 0
	for i := 1; i <= n; i++ {
		r += i * i
	}
	return r
}

func Difference(n int) int {
	return SquareOfSums(n) - SumOfSquares(n)
}
